### DISCUSSION TITLE
[ReverseProxyPlugin] REVERSE_PROXY_PASS to HTTPS upstream

### DISCUSSION THREAD

**Discussion by zeevikhippo** [abhinavsingh/proxy.py#636] [Q&A] [answered]
I'm trying to run reverseProxy with https upstream connection:
INITIATOR <== http/https ==>ReversePROXY <== https ==> upstream target.

I tried setting the REVERSE_PROXY_PASS to be 'https://<my_https_serevr'.
I also tried to set the default port to be 443,
Tried starting it with/out certificates, wiyh/out enableing the web server.
Nothing works, it keeps reversing it as HTTP request

Is it supported?
Any tips or hints how to accomplish that?

Thanks,
Zeevik

And, yes. Technical reasons require me to run this exact setup :)

**Comment by abhinavsingh** [comment]
@zeevikhippo https://github.com/abhinavsingh/proxy.py/pull/638 has just landed, it adds ton of improvements.  I'll also give your mentioned workflow a try over the weekend.  IMHO, this should work, if not out of box, with a little configuration.

**Reply by zeevikhippo** [reply]
Thjanks @abhinavsingh .
I just pulled the latest code and gave it a try.
Looks like it does not cross HTTP/HTTPS - if I send HTTP, it will proxy that as HTTP

**Reply by abhinavsingh** [reply]
Yes, that PR still doesn't contain fix for reverse proxy.  But the core feature needed to support it has landed.  I'll take sometime to update `ReverseProxy` plugin.  You can also check `proxy_pool.py` on how to make upstream `HTTPS` requests from within a plugin.

**Reply by abhinavsingh** [reply]
Update: `ReverseProxy` and `ProxyPool` both are now update to work `async` and with `https`.  Please give it a try.

**Comment by abhinavsingh** [comment, marked as answer]
Fixed with https://github.com/abhinavsingh/proxy.py/pull/777
